utils/json/height.go

```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package json
+
+import "strconv"
+
+type Height struct {
+	Numeric    Uint64
+	IsProposed bool
+}
```

Comment:
Similarly to Cey's comments, I wonder if rather than using a struct for this if it would be nicer to have special values for `uint64`.

Specifically - `type Height uint64`

- When json marshaling - `Height == MaxUint64` gets mapped to `proposed`
- When json unmarshaling - `proposed` gets mapped to `MaxUint64`
- When json unmarshaling - `MaxUint64` is considered invalid

That way in the client we could just pass in `api.ProposedHeight` rather than adding a new bool arg.

(Also I think this should probably live in the `api` package rather than the `json` package, this feels more API specific than general to json helpers)